src/lib.rs

```diff
@@ -534,6 +535,36 @@ impl<T: Abomonation> Abomonation for Box<T> {
     }
 }
 
+impl<T: Abomonation + 'static + Sized + Unpin> Abomonation for Arc<T> {
+    #[inline]
+    unsafe fn entomb<W: std::io::Write>(&self, write: &mut W) -> std::io::Result<()> {
+        /* Since the value T has not yet been seen by abomonate (it is behind a pointer)
+         * we need to fully encode it.
+         */
+        encode(self.as_ref(), write)
+    }
+    #[inline]
+    unsafe fn exhume<'a, 'b>(&'a mut self, bytes: &'b mut [u8]) -> Option<&'b mut [u8]> {
+        use std::ptr;
+        /* The idea here is to construct a new Arc<T> from the entombed bytes.
+         * The state of this ArcVal upon entry of this function contains only an invalid
+         * pointer to an ArcInner that we need to dispose of without trying to run
+         * its destructor (which would panic).
+         */
+        let (value, bytes) = decode::<T>(bytes)?;
```

Comment:
@frankmcsherry Here `value` contains a reference to a `T`. Since `T: 'static` this structure cannot contain non-static references (but it could contain boxes). My understanding is that T can only directly reference `bytes` by unsafely reinterpreting a reference as a box or equivalently by transmuting a lifetime to `'static`. Is my assumption correct that such behaviour violates the contract of Abomonation?